File: NumberFormat.cs
using System;

public class Kata
{
  public static string NumberFormat(int number)
  {
    var numberFormat = function (number) 
    {
      return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
    };
  }
}

using NUnit.Framework;

[TestFixture]
public class Tests
{
  [Test]
  [TestCase(100000, ExpectedResult="100,000")]
  [TestCase(5678545, ExpectedResult="5,678,545")]
  public static string FixedTest(int num)
  {
    return Kata.NumberFormat(num);
  }
}

File: numberFormat.js
/*
Format any integer provided into a string with "," (commas) in the correct places.

Example:

For n = 100000 the function should return '100,000';
For n = 5678545 the function should return '5,678,545';
for n = -420902 the function should return '-420,902'.
*/

var numberFormat = function (number) {
  return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
};

const Test = require('@codewars/test-compat');

describe("Tests", () => {
  it("test", () => {
    Test.assertEquals(numberFormat(100000), '100,000');
    Test.assertEquals(numberFormat(5678545), '5,678,545');
  });
});